src/main/java/com/develhope/spring/deals/components/mappers/RentalMapper.java

```diff
@@ -25,30 +25,35 @@ public class RentalMapper {
     public Rental toEntity(RentalCreatorDTO rentalCreatorDTO) {
         Vehicle vehicle = new Vehicle(rentalCreatorDTO.getVehicleId());
         User user = new User(rentalCreatorDTO.getUserId());
+        User seller = new User(rentalCreatorDTO.getSellerId());
         Rental rental = new Rental();
         rental.setStartDate(rentalCreatorDTO.getStartDate());
         rental.setEndDate(rentalCreatorDTO.getEndDate());
         rental.setPaid(rentalCreatorDTO.isPaid());
         rental.setVehicle(vehicle);
         rental.setUser(user);
+        rental.setSeller(seller);
         return rental;
     }
 
     public Rental toEntity(RentalUpdaterDTO rentalUpdaterDTO) {
         Vehicle vehicle = new Vehicle(rentalUpdaterDTO.getVehicleId());
         User user = new User(0);
+        User seller = new User(0);
         Rental rental = new Rental();
         rental.setStartDate(rentalUpdaterDTO.getStartDate());
         rental.setEndDate(rentalUpdaterDTO.getEndDate());
         rental.setPaid(rentalUpdaterDTO.isPaid());
         rental.setUser(user);
+        rental.setSeller(seller);
         rental.setVehicle(vehicle);
         return rental;
     }
 
     public RentalReturnerDTO toReturnerDTO(Rental rental) {
         VehicleRentalReturnerDTO vehicleRentalReturnerDTO = vehicleMapper.toRentalReturnerDTO(rental.getVehicle());
-        BuyerRentalReturnerDto buyerReturnerDto = userMapper.toRentalBuyerDTO(rental.getUser());
+        UserRentalReturnerDto buyerReturnerDto = userMapper.toUserBuyerDTO(rental.getUser());
+        UserRentalReturnerDto sellerReturnerDTO = userMapper.toUserBuyerDTO(rental.getSeller());
```

Comment:
A questo punto sarebbe il caso di cambiare nome del DTO in modo che sia generico

Reply:
Infatti avevo cambiato il nome della classe, ma ho dimenticato di cambiare il nome al metodo e alla variabile, lmao

Reply:
Il nome della classe del DTO è stato cambiato in User così da essere generico, le variabili hanno, invece, i nomi di questi oggetti sono, invece, specifici per i ruoli

Reply:
Il nome del metodo, invece è da cambiare, quello sì